jetty-rewrite/src/main/java/org/eclipse/jetty/rewrite/handler/CompactPathRule.java

```diff
@@ -35,7 +35,7 @@ public CompactPathRule()
     @Override
     public void applyURI(Request request, String oldURI, String newURI) throws IOException
     {
-        String uri = request.getRequestURI();
+        String uri = oldURI;
         if (uri.startsWith("/"))
             uri = URIUtil.compactPath(uri);
```

Comment:
Reminder, the `URIUtil.compactPath(String)` API preserves the query section of the input string, and only operates on the path portion.